charmcraft.yaml

```diff
@@ -10,7 +10,7 @@ parts:
 bases:
   - build-on:
     - name: "ubuntu"
-      channel: "20.04"
+      channel: "22.04"
     run-on:
     - name: "ubuntu"
-      channel: "20.04"
+      channel: "22.04"
```

Comment:
I think we should be careful about changing this. Doing series upgrades for machine charms is non-trivial, and means anyone running the current charm on focal won't see there's a new version available.

Reply:
The problem is it is not possible to build the charm on 20.04 any more.

Doing `charmcraft pack` will result in the following error:
```
2022-12-14 09:16:40.188 :: 2022-12-14 01:16:39.876 package: rust-all
2022-12-14 09:16:40.188 :: 2022-12-14 01:16:39.879 Parts processing error: Package 'rust-all' has unmet dependencies: llvm-14.
```

This charm requires `rust-all` package.

On focal, the dependency `llvm-14` is not available:
https://packages.ubuntu.com/focal/rust-all

It is possible to install `rust-all` on jammy:
https://packages.ubuntu.com/jammy/rust-all

The requirement of `rust-all` stems from the python package `cryptography` which is needed by the python package `pylxd`.

Either we remove the dependency on `pylxd` or find a different way to manage the rust dependency (I already put this in the github runner design doc improvement list).

Reply:
If it's possible to build on jammy and run on focal, that sounds good to me.

Could we add integration tests for focal and jammy? It looks like the (very basic) integration tests currently just test it deploys and goes into blocked state, and this would presumably be on the default series (which you're switching to jammy here) if I'm understanding it correctly.

Reply:
The integration tests are the next Jira story

Reply:
Ok, thanks. If you've manually tested it in the meantime I think that's fine for now - the other option would be to keep the default (based on ordering of the series entries in metadata.yaml) as focal for now so we'd ensure it can continue to be deployed on focal until we get the integration tests in place (obviously we can also just choose not to publish to stable until we have those tests in place). But in any case I'll close out this comment thread.